subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/IndexPageGenerator.java

```diff
@@ -118,4 +122,16 @@ public void render(final ResultsStore store, Writer writer) throws IOException {
             endAll();
         }};
     }
+
+    private List<? extends PerformanceTestExecution> filterForRequestedCommit(List<? extends PerformanceTestExecution> results) {
+        if (commitId == null) {
+            return results;
+        }
+        for (PerformanceTestExecution execution : results) {
+            if (execution.getVcsCommits().contains(commitId)) {
```

Comment:
I wanted to do that at first. But `ResultsStore`is an interface which is implemented several times. It feels like there is some redundancy/unnecessary complexity, which we would increase. So I decided for this simpler (in terms of code complexity) solution. And as you wrote, there are only a few entries to check.
On the long run, if we are going to maintain and extend this performance test tooling code, we should do more simplifications and might change this then.